
## V povprečju, koliko je poškodb, ali žrtev v prometnih nesrečah, in če je to število povečano v nesrečah povezanih z alkoholom.

In [20]:
# Relevantni importi
% matplotlib inline
import matplotlib.image as mpimg
import matplotlib.pyplot as plt
from Orange.data.filter import SameValue
from Orange.data import Table
import numpy as np
from csv import DictReader

In [21]:
# Filtrirani podatki naloženi (crash data filt je isti, ker je tako ali tako cel)
crash1_reader = DictReader(open('data/Crash_Data_1_filt.csv', 'rt', encoding='utf-8'))
crash2_reader = DictReader(open('data/Crash_Data_2_filt_non_rand.csv', 'rt', encoding='utf-8'))
liquor_reader = DictReader(open('data/Liquor_Data_filt_non_rand.csv', 'rt', encoding='utf-8'))

In [22]:
# Pregledam kako vnosi podatkov zgledajo
for crash in crash1_reader:
    print(crash)
    break
print("------------------")
for crash in crash2_reader:
    print(crash)
    break
print("------------------")
for liquor in liquor_reader:
    print(liquor)
    break

OrderedDict([('\ufeffX', '-93.511904910700665'), ('Y', '41.648063161780378'), ('OBJECTID', '1001'), ('CRASH_KEY', '2016902721'), ('CASENUMBER', '2016902721'), ('LECASENUM', '16-452'), ('CRASH_DATE', '2016-01-20T00:00:00.000Z'), ('CRASH_MONTH', '1'), ('CRASH_DAY', '4'), ('TIMESTR', '20:02'), ('DISTRICT', '1'), ('COUNTY_NUMBER', '77'), ('CITYNAME', '132'), ('SYSTEMSTR', 'US 6'), ('LITERAL', 'US 6/HUBBELL AVE/NE RAMP/SPEC CASE/FREDERICK M HUBBELL AVE'), ('FRSTHARM', '33'), ('LOCFSTHRM', '1'), ('CRCOMNNR', '4'), ('MAJCSE', '9'), ('DRUGALC', '8'), ('ECNTCRC', '1'), ('LIGHT', '5'), ('CSRFCND', '1'), ('WEATHER', '1'), ('RCNTCRC', '1'), ('RDTYP', '22'), ('PAVED', '1'), ('WZRELATED', ''), ('CSEV', '5'), ('FATALITIES', '0'), ('INJURIES', '0'), ('MAJINJURY', '0'), ('MININJURY', '0'), ('POSSINJURY', '0'), ('UNKINJURY', '0'), ('PROPDMG', '6000'), ('VEHICLES', '2'), ('TOCCUPANTS', '2'), ('REPORT', '7'), ('XCOORD', '457373'), ('YCOORD', '4610828')])
------------------
OrderedDict([('\ufeffX', '-92.65

In [25]:
# Povprečno število mrtvih v vseh prometnih nesrečah
crash1_reader = DictReader(open('data/Crash_Data_1_filt.csv', 'rt', encoding='utf-8'))
crash2_reader = DictReader(open('data/Crash_Data_2_filt_non_rand.csv', 'rt', encoding='utf-8'))

crash1_fatality_sum = 0
crash1_fatality_count = 0

crash2_fatality_sum = 0
crash2_fatality_count = 0
for crash in crash1_reader:
    crash1_fatality_sum += int(crash["FATALITIES"])
    crash1_fatality_count +=1

for crash in crash2_reader:
    crash2_fatality_sum += int(crash["FATALITIES"])
    crash2_fatality_count +=1
print("Povprečje v crash1:")
print(str(crash1_fatality_sum/crash1_fatality_count))
print("Povprečje v crash2:")
print(str(crash2_fatality_sum/crash2_fatality_count))

Povprečje v crash1:
0.007392870507645679
Povprečje v crash2:
0.00793852570236956


** Torej verjetnost smrti v vseh vnosih je zelo mala **
#### (crash1 povp = 0.007392870507645679 ) 
#### (crash2 povp = 0.00793852570236956 ) 

** Pogledal bom koliko je povprečno poškodb (večjih in manjših) v vseh prometnih nesrečah ** 

In [42]:
# Povprečno število poškodb v vseh prometnih nesrečah
crash1_reader = DictReader(open('data/Crash_Data_1_filt.csv', 'rt', encoding='utf-8'))
crash2_reader = DictReader(open('data/Crash_Data_2_filt_non_rand.csv', 'rt', encoding='utf-8'))

crash1_fatality_sum = 0
crash1_fatality_count = 0

crash2_fatality_sum = 0
crash2_fatality_count = 0
for crash in crash1_reader:
    crash1_fatality_sum += int(crash["INJURIES"])
    crash1_fatality_count +=1


print("Povprečje v crash1:")
print(str(crash1_fatality_sum/crash1_fatality_count))

Povprečje v crash1:
0.3853776562772269


###### Te podatke ima samo zbirka crash1, in sicer je povprečje 0.3853776562772269

** Spodaj bom preveril verjetnost poškodbe in smrti v zbirki crash1 **

In [28]:
# Povprečno število vnosov z poškodbami in smrtmi v crash1
crash1_reader = DictReader(open('data/Crash_Data_1_filt.csv', 'rt', encoding='utf-8'))

crash1_fatality_count = 0
crash1_injury_count = 0
crash1_count = 0

for crash in crash1_reader:
    if(int(crash["INJURIES"])>0):
       crash1_injury_count += 1 
    if(int(crash["FATALITIES"])>0):
       crash1_fatality_count += 1
    crash1_count +=1


print("Verjetnost vnosa z smrtjo v podatkih:")
print(str(crash1_fatality_count/crash1_count))
print("Verjetnost vnosa z poškodbo v podatkih:")
print(str(crash1_injury_count/crash1_count))

Verjetnost vnosa z smrtjo v podatkih:
0.006630159369492126
Verjetnost vnosa z poškodbo v podatkih:
0.285131132040438


Verjetnost vnosa z smrtjo v podatkih:  
0.006630159369492126  
Verjetnost vnosa z poškodbo v podatkih:  
0.285131132040438

#### Sedaj bom preveril če so te vrednosti povečane pri nesrečah povezanih z alkoholom

In [36]:
# Povprečno število mrtvih v vseh prometnih nesrečah
crash2_reader = DictReader(open('data/Crash_Data_2_filt_non_rand.csv', 'rt', encoding='utf-8'))

crash2_fatality_sum = 0
crash2_fatality_count = 0

for crash in crash2_reader:
    if(float(crash["ALCRESULT"])>0):
        crash2_fatality_sum += int(crash["FATALITIES"])
        crash2_fatality_count +=1

print("Povprečje smrti v crash2 kjer je alcresult večji od 0:")
print(str(crash2_fatality_sum/crash2_fatality_count))

Povprečje v crash2:
0.05073869571705716


** Povprečje smrti v vseh vnosih z pozitivnim alkohol testom je 0.05073869571705716 **


In [43]:
# Povprečno število smrti za nesreče z alkohol testom > 0
crash2_reader = DictReader(open('data/Crash_Data_2_filt_non_rand.csv', 'rt', encoding='utf-8'))
crash2_fatality_count = 0
crash2_count = 0

for crash in crash2_reader:
    if(int(crash["FATALITIES"])>0 and float(crash["ALCRESULT"])>0):
       crash2_fatality_count += 1
    if float(crash["ALCRESULT"])>0:
        crash2_count +=1


print("Verjetnost vnosa z smrtjo, kjer je alkohol test pozitiven:")
print(str(crash2_fatality_count/crash2_count))

Verjetnost vnosa z smrtjo, kjer je alkohol test pozitiven:
0.043277122817489924


Verjetnost vnosa z smrtjo, kjer je alkohol test pozitiven je torej  0.0008284175795923614

| Vse | Povp št. mrtvih | Povp št. poškodovanih | Verjetnost vnosa z smrtjo | Verjetnost vnosa z poškodbo |
|-----|-----|-----|-----|-----|
| | 0.00793852570236956 | 0.3853776562772269   | 0.006630159369492126  | 0.285131132040438  |

| Alk. test > 0 | Povp št. mrtvih | Povp št. poškodovanih | Verjetnost vnosa z smrtjo | Verjetnost vnosa z poškodbo |
|-----|-----|-----|-----|-----|
| | 0.05073869571705716 |  //  | 0.043277122817489924  | //  |



** Vidimo, da sta vrednosti povečane, kjer smo upoštevali samo vnose z pozitivnim alk. testom, ampak, ker ima samo druga podatkovna zbirka podatek o alkohol testu, nima pa podatkov o poškodovanih, tega nemoremo primerjati **
